### DISCUSSION TITLE
Task Stuck in Queued Status after Dag Recovered

### DISCUSSION THREAD

**Discussion by enamespace** [apache/airflow#31523] [Q&A]
### Apache Airflow version

2.2.5

### What happened

I have 2 task in 1 dag,  when first task became queued status , I changed the dag code and led to import error, then the first task finished and the second became queue status. but when i recovered the dag and no import error. the second task still kept queued status. after two schedule interval went by, it still kept queued status and no next dagrun.
<img width="614" alt="image" src="https://github.com/apache/airflow/assets/132480235/08581857-da2d-480b-bc78-3d67eadba0a6">
<img width="526" alt="image" src="https://github.com/apache/airflow/assets/132480235/2af09284-dd19-45e2-b680-9487a8f835e0">
<img width="649" alt="image" src="https://github.com/apache/airflow/assets/132480235/9869e559-4a4c-4143-8982-45e751fd860e">
<img width="923" alt="image" src="https://github.com/apache/airflow/assets/132480235/b046b724-6eac-406a-85cf-cce143e4b6b9">

### What you think should happen instead

I thought the second task would become running when the dag recovered.

### How to reproduce

like above description

### Operating System

Debian GNU/Linux 10 (buster)

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon          3.2.0
apache-airflow-providers-apache-hive     2.3.2
apache-airflow-providers-celery          2.1.3
apache-airflow-providers-cncf-kubernetes 3.0.0
apache-airflow-providers-databricks      2.5.0
apache-airflow-providers-ftp             2.0.1
apache-airflow-providers-http            2.1.2
apache-airflow-providers-imap            2.2.0
apache-airflow-providers-jdbc            2.1.3
apache-airflow-providers-mysql           2.2.3
apache-airflow-providers-postgres        4.1.0
apache-airflow-providers-presto          2.1.2
apache-airflow-providers-slack           4.2.3
apache-airflow-providers-sqlite          2.1.0
apache-airflow-providers-ssh             2.4.3

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

helm deployment for airflow 2.2.5

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by eladkal** [comment]
Please update to latest Airflow.
Airflow 2.6.0 did a lot of refactoring/changes to address the task stuck in queue status issue.

**Reply by enamespace** [reply]
Thanks, I will try later. 
Can you give a simple explanation or related issue about the situation? I wonder why it became queued and not running. I checked the the postgresql and found the db has queued task information and running dagrun information but not found in redis "unack" key and any task log on worker. I learned from 
https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/celery.html and https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/tasks.html  but still confused about the task state transition between  worker, redis, postgresql and executor like where the state transition happen and who change the state. Could you help to give some suggestions or links about how to learn airflow workflow?
